packages/core/admin/admin/src/content-manager/components/BlocksEditor/utils/links.js

```diff
@@ -0,0 +1,89 @@
+import { Transforms, Editor, Element as SlateElement, Node } from 'slate';
+
+/**
+ *
+ * @param {string} url
+ * @param {string} protocol
+ */
+const addProtocol = (url, protocol = 'https://') => {
+  const allowedProtocols = ['http://', 'https://', 'mailto:', 'tel:'];
+
+  if (allowedProtocols.some((allowedProtocol) => url.startsWith(allowedProtocol))) {
+    return url;
+  }
+
+  return `${protocol}${url}`;
+};
+
+/**
+ *
+ * @param {import('slate').Editor} editor
+ */
+const removeLink = (editor) => {
+  Transforms.unwrapNodes(editor, {
+    match: (node) => !Editor.isEditor(node) && SlateElement.isElement(node) && node.type === 'link',
+  });
+};
+
+/**
+ *
+ * @param {import('slate').Editor} editor
+ * @param {object} link
+ * @param {string} link.url
+ */
+const insertLink = (editor, { url }) => {
+  if (editor.selection) {
+    // We want to remove all link on the selection
+    // We use a try catch because Slate throw error when there is no link in the selection
```

Comment:
I removed the try catch but unable to reproduce the error:
- I select some text not containing a link
- I click the link button
- Add a link
- Hit save, and it works
Maybe I'm missing something. 

I'm worried we could end up silencing an error that would be difficult to track down if we ever actually have a bug